src/olmo_core/train/trainer.py

```
@@ -1056,6 +1064,28 @@ def persist_working_file(self, name: PathOrStr) -> PathOrStr:
             raise FileNotFoundError(source)
         return target
 
+    def persist_working_subdir(self, name: PathOrStr) -> PathOrStr:
+        """
+        Persist a directory in the :data:`work_dir` by saving/uploading it to the :data:`save_folder`.
+
+        :param name: The name/path of the directory relative to the :data:`work_dir`.
+
+        :returns: The full path/URL to the saved directory.
+
+        :raises FileNotFoundError: If the directory can't be found.
+        :raises FileExistsError: If any files in the directory already exist in the save folder and :data:`save_overwrite`
+            is ``False``.
+        """
+        if Path(name).is_relative_to(self.work_dir):
+            name = Path(name).relative_to(self.work_dir)
+        source = join_path(self.work_dir, name)
+        target = join_path(self.save_folder, name)
+        if source != target:
+            copy_dir(source, target, save_overwrite=self.save_overwrite)
+        elif dir_is_empty(source):
```

`dir_is_empty()` returns True when the directory is empty or does not exist, which matches our requirement since the directory should exist and not be empty if `source == target`. I considered creating a `dir_exists()` function to align better with `persist_working_file()`'s implementation, but 1) this function should raise `FileNotFoundError` in both cases (if the directory is missing or exist but is empty), and 2) If i were to create a standalone `dir_exists()` function in `io.py`, I think it should return True when the dir exists but is empty, which we don't want here.